tests/workload/network.go

```diff
@@ -45,11 +50,24 @@ func (d DefaultTestNetworkConfiguration) Parser() chain.Parser {
 	return d.parser
 }
 
+func (d DefaultTestNetworkConfiguration) AuthFactories() []chain.AuthFactory {
+	authFactories := make([]chain.AuthFactory, 0, len(d.keys))
+	for _, key := range d.keys {
+		authFactories = append(authFactories, auth.NewED25519Factory(key))
+	}
+	return authFactories
+}
```

Comment:
Can we remove the need for this by just setting the field to be an `AuthFactory` from the start?